warehouse/query-core/src/test/java/datawave/query/jexl/visitors/PushdownNegationVisitorTest.java

```diff
@@ -280,6 +287,30 @@ public void testApplyDeMorgansNegateOrRoot() {
         children.add(child2);
         JexlNode or = JexlNodeFactory.createUnwrappedOrNode(children);
         JexlNode result = PushdownNegationVisitor.applyDeMorgans(or, true);
-        Assert.assertEquals("!((!(f1 == 'v1') && !(f2 == 'v2')))", JexlStringBuildingVisitor.buildQuery(result));
+        Assert.assertEquals("!(!(f1 == 'v1') && !(f2 == 'v2'))", JexlStringBuildingVisitor.buildQuery(result));
+    }
+    
+    @Test
+    public void test_doubleWrappingUnionAfterPushdown() throws ParseException {
+        String query = "!(FOO == 'bar' || FOO == 'baz')";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        ASTJexlScript result = (ASTJexlScript) PushdownNegationVisitor.pushdownNegations(script);
+        
+        // Really? Double wrapping? That's very untidy and this is very passive aggressive.
```

Comment:
I am not getting your passive aggressive comment here.  Please clean this up.

Reply:
I'll get rid of that